lib/src/broker/broker_node_provider.dart

```diff
@@ -470,10 +544,16 @@ class BrokerNodeProvider extends NodeProviderImpl implements ServerLinkManager {
     tokens.children.forEach((String name, TokenGroupNode node) {
       m[name] = node.serialize(true);
     });
-    File connsFile = new File("tokens.json");
-    if (shouldSaveFiles) {
-      await connsFile.writeAsString(DsJson.encode(m));
+
+    if (saveHandler != null) {
+      await saveHandler("tokens", m);
+    } else {
+      File connsFile = new File("tokens.json");
```

Comment:
I feel like we reference files a lot, and we re-instantiate them many times.

Could they be extracted?